### PR TITLE
#1965-introduce functions to allow speech to be muted without tapping button

### PR DESCRIPTION
**Pull request by abhishek1508**
## Description

**Feature:**
To allow the user to start the navigation in muted state without tapping the toggle mute button.

### Goal

Add a new function that allows user to use the speech player to mute/unmute the navigation instructions.

### Implementation

I see that there is a function `toggleMute` in `SoundButton` class. I think this should not be publicly available, as it just toggles the UI of the mute/unmute button if accessed by code. It's only when the user taps on the button, that the mute/unmute effect takes place. Look at lines 517-521 of `InstructionView`. 

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: app/src/androidTest/java/testapp/NavigationViewTest.java [hunk 1]
```diff
@@ -3,54 +3,100 @@
...
+import com.mapbox.services.android.navigation.ui.v5.listeners.SpeechAnnouncementListener;
 import com.mapbox.services.android.navigation.ui.v5.map.NavigationMapboxMap;
+import com.mapbox.services.android.navigation.ui.v5.voice.SpeechAnnouncement;
 import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation;
-
 import org.junit.Test;
-
 import testapp.activity.BaseNavigationActivityTest;
 
-import static junit.framework.Assert.assertNotNull;
```
Comment (Danm72) [inline app/src/androidTest/java/testapp/NavigationViewTest.java]: Can you change this back from a wildcard import?

File: app/src/androidTest/java/testapp/NavigationViewTest.java [hunk 2]
```diff
@@ -3,54 +3,100 @@
...
+import static junit.framework.Assert.*;
 import static testapp.action.NavigationViewAction.invoke;
 
-public class NavigationViewTest extends BaseNavigationActivityTest {
+public class NavigationViewTest extends BaseNavigationActivityTest implements SpeechAnnouncementListener {
+
+    @Override
```
Comment (Danm72) [inline app/src/androidTest/java/testapp/NavigationViewTest.java]: I think your codestyles are mismatched with the project hense all the line changes here, you might want to fix that and resubmit

**Comment by Danm72** [comment]
I agree that this feature would be helpful

**Comment by Danm72** [comment]
@abhishek1508 I was able to mute in the short term by using a shared preference and overriding willVoice from SpeechAnnouncementListener

` override fun willVoice(announcement: SpeechAnnouncement?): SpeechAnnouncement? {
        return if (isMuted) {
            announcement
        } else {
            null
        }
    }`

**Comment by Guardiola31337** [comment]
Is this something currently supported in `1.0`? Could you confirm @abhishek1508?

cc @JunDai

**Comment by JunDai** [comment]
https://github.com/mapbox/mapbox-navigation-android/pull/3283 is opened for `v1.0`

**Pull request closed without merging**